File: screen-switcher-sample/src/main/java/com/jaynewstrom/screenswitchersample/MainActivityComponent.java
package com.jaynewstrom.screenswitchersample;

import com.jaynewstrom.screenswitchersample.base.ParentComponent;

import dagger.Component;

@ForMainActivity
@Component(
        dependencies = {
                ApplicationComponent.class
        },
        modules = {
                MainActivityModule.class
        }
)
public interface MainActivityComponent extends ParentComponent {
    void inject(MainActivity mainActivity);
}

File: screen-switcher-sample/src/main/java/com/jaynewstrom/screenswitchersample/activity/MainActivityComponent.kt
package com.jaynewstrom.screenswitchersample.activity

import com.jaynewstrom.screenswitcher.dialogmanager.DialogManagerComponent
import com.jaynewstrom.screenswitchersample.application.ApplicationComponent
import com.jaynewstrom.screenswitchersample.core.ScreenParentComponent
import dagger.Component

@ForMainActivity
@Component(dependencies = [ApplicationComponent::class], modules = [MainActivityModule::class])
internal interface MainActivityComponent : ScreenParentComponent, DialogManagerComponent {
    fun inject(mainActivity: MainActivity)
}
